File: solutions/1347. Minimum Number of Steps to Make Two Strings Anagram/1347.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
public:
    int minSteps(string_view s, string_view t) {
        vector<int> count(26);

        for (const char c : s)
            ++count[c - 'a'];

        for (const char c : t)
            --count[c - 'a'];

        return accumulate(count.begin(), count.end(), 0,
                          [](int subtotal, int c) { return subtotal + abs(c); }) / 2;
    }
};

File: solutions/1347. Minimum Number of Steps to Make Two Strings Anagram/1347.rs
// Time complexity: O(n)
// Space complexity: O(1)
impl Solution {
    pub fn min_steps(s: String, t: String) -> i32 {
        let mut counts = [0; 26];

        for byte in s.bytes() {
            counts[(byte - b'a') as usize] += 1;
        }

        for byte in t.bytes() {
            counts[(byte - b'a') as usize] -= 1;
        }

        counts.iter().map(|&count: &i32| count.abs()).sum::<i32>() >> 1
    }
}